### DISCUSSION TITLE
Monster killed at the same time do not trig their common event on death.

### DISCUSSION THREAD

**Discussion by Spectreye** [AscensionGameDev/Intersect-Engine#1498] [Ideas]
**Describe the bug**
Monster killed at the same time do not trig their common event on death, it can lead to event being inactive because of a wrong count. See the video for a demonstration.

**Version Encountered**
0.6.2.448

**To Reproduce**
1. Create a global variable  "Count", create a common event "Deathcount" wich lower the variable "Count" by 1, you can also put a chatbox message wich tell the actual count using a \gv{#}  .
2. Create a NPC, "Skeleton" (or whatever) Give him the "Deathcount" common event.
3. Make an event wich spawn let's say, 9 skeleton and set the variable "Count" at 9, and as condition the "Count" must be equal to 0.
4. Make an Event "Skeleton Counter" who will tell you the actual count of skeleton killed using a \gv{#}.
4. Use an AOE spell wich can kill skeleton together at the same time.
5. Talk back to the Skeleton Counter See error.

**Expected behavior**
Each death of monster should trigger the common event.

**Video**
https://s3.us-east-2.amazonaws.com/ascensiongamedev/filehost/c52d176f7835a3ad9c865d42a725c61d.mp4

**Logs**
X

**Operating System:**
Windows

**Comment by jcsnider** [comment]
So this is technically working by design. Events are set up so online 1 instance of an event can ever run at one time. I do, however, concede that the intended functionality does not work well for this use case.

Not sure when but I'll likely attempt a fix for this at a later point.

**Comment by WeylonSantana** [comment]
Added in #2269
